src/connection.js

```diff
@@ -39,6 +39,12 @@ class Connection extends EventEmitter {
     super();
 
     this.config = config;
+
+    if (typeof (config.domain) === 'string') {
+      this.config = JSON.parse(JSON.stringify(config));
```

Comment:
I'll and send an update to the pull request with comments added... once I figure how :)